packages/astro/src/assets/services/service.ts

```diff
@@ -229,7 +229,7 @@ export const baseService: Omit<LocalImageService, 'transform'> = {
 
 		const aspectRatio = targetWidth / targetHeight;
 		const imageWidth = isESMImportedImage(options.src) ? options.src.width : options.width;
-		const maxWidth = imageWidth ?? Infinity;
+		const maxWidth = isESMImportedImage(options.src) ? imageWidth ?? Infinity : Infinity;
```

Comment:
nit: we already check `sESMImportedImage(options.src)` at line 231. We could save the result of this check in a local variable and use it in your new code.

Reply:
If you save it in a local variable, the type inference doesn't work

Reply:
TypeScript.....

Reply:
Yeah, it's done 3 times in that function call... I had it as `isLocal` variable initially, but then had to revert it back to this.